src/models/dirwatcher.ts

```diff
@@ -0,0 +1,77 @@
+import EventEmitter from "events";
+import fs  from "fs";
+import _ from "lodash";
+import { promisify } from "util";
+
+const readDir = promisify(fs.readdir);
+const stat = promisify(fs.stat);
+
+enum FileStatus {
+  ADDED = "added",
+  MODIFIED = "modified",
+  DELETED = "deleted",
+ }
+
+export class DirWatcher extends EventEmitter {
+  timerId: any;
+  files: string[];
+  fileTempStatistics: object;
+  fileStatistics: object;
+
+  constructor() {
```

Comment:
Please, consider distinguishing the descriptors like `private` and `public`
1. all methods and properties of class that are used only inside this class should have `private` descriptor
2. all other properties and methods should have `public` explicitly
3. you can define the properties outside the constructor like `private files: string[] = []` and so on, no need to set them in constructor